op-acceptor/reporting/tree_formatters.go

```diff
@@ -0,0 +1,607 @@
+			return i == len(visibleSiblings)-1
+		}
+	}
+	return true
+}
+
+// getStatusChar returns a character representing the test status
+func (f *TreeTextFormatter) getStatusChar(status types.TestStatus) string {
+	switch status {
+	case types.TestStatusPass:
+		return "✓"
+	case types.TestStatusFail:
+		return "✗"
+	case types.TestStatusSkip:
+		return "⊝"
+	case types.TestStatusError:
+		return "⚠"
+	default:
+		return "?"
+	}
+}
+
+// TreeJSONFormatter formats test trees as JSON
+type TreeJSONFormatter struct {
+	includeTestResults bool
+	includeHierarchy   bool
+}
+
+// NewTreeJSONFormatter creates a new tree-based JSON formatter
+func NewTreeJSONFormatter(includeTestResults, includeHierarchy bool) *TreeJSONFormatter {
+	return &TreeJSONFormatter{
+		includeTestResults: includeTestResults,
+		includeHierarchy:   includeHierarchy,
+	}
+}
+
+// Format formats a test tree as JSON
+func (f *TreeJSONFormatter) Format(tree *types.TestTree) (string, error) {
+	data := make(map[string]interface{})
+
+	// Basic tree information
+	data["runId"] = tree.RunID
+	data["networkName"] = tree.NetworkName
+	data["timestamp"] = tree.Timestamp
+	data["duration"] = tree.Duration
+	data["stats"] = tree.Stats
+
+	// Include hierarchy if requested
+	if f.includeHierarchy {
+		data["hierarchy"] = f.nodeToJSON(tree.Root)
+	}
+
+	// Include flat test list
+	var tests []map[string]interface{}
+	for _, node := range tree.TestNodes {
+		testData := map[string]interface{}{
+			"id":             node.ID,
+			"name":           node.Name,
+			"type":           node.Type,
+			"status":         node.Status,
+			"duration":       node.Duration,
+			"executionOrder": node.ExecutionOrder,
+			"package":        node.Package,
+			"gate":           node.Gate,
+			"suite":          node.Suite,
+			"depth":          node.Depth,
+			"path":           node.GetPath(),
+		}
+
+		if node.Error != nil {
+			testData["error"] = node.Error.Error()
+		}
+
+		if node.LogPath != "" {
+			testData["logPath"] = node.LogPath
+		}
+
+		// Include original test result if requested
+		if f.includeTestResults && node.TestResult != nil {
+			testData["testResult"] = node.TestResult
+		}
+
+		tests = append(tests, testData)
+	}
+	data["tests"] = tests
+
+	// Include failed tests summary
+	var failed []string
+	for _, node := range tree.FailedNodes {
+		failed = append(failed, node.GetPath())
+	}
+	data["failedTests"] = failed
+
+	// Convert to JSON
+	jsonBytes, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal JSON: %w", err)
+	}
+
+	return string(jsonBytes), nil
+}
+
+// nodeToJSON converts a tree node to JSON representation
+func (f *TreeJSONFormatter) nodeToJSON(node *types.TestTreeNode) map[string]interface{} {
+	nodeData := map[string]interface{}{
+		"id":       node.ID,
+		"name":     node.Name,
+		"type":     node.Type,
+		"status":   node.Status,
+		"duration": node.Duration,
+		"depth":    node.Depth,
+	}
+
+	if node.Package != "" {
```

Comment:
I'm not sure I completely understand all this. Presumably there's not a whole lot of difference between an  absent key and an empty string, so why are we making one here? (and above)

Wouldn't it be clearer to have a proper object to marshal (with the right "omitempty" and whatnot), vs having those maps with optional items?